### DISCUSSION TITLE
Should we enforce SP ids to be unique?

### DISCUSSION THREAD

**Discussion by t-ober** [ie3-institute/simona#346] [General]
Currently system participant ids have to be unique as they are used for actor naming where they have to be unique. Since we don't enforce this in the PSDM we should think about if this makes sense.

**Comment by sebastian-peter** [comment]
I see arguments for and against enforcing uniqueness. On one hand, I don't really care that much as long as the UUIDs are unique. On the other hand, the id is still an id, which means that one should be able to identify the object just by it. If decide that we want to drop the requirement, we should think about renaming it, maybe to "name".

**Comment by staudtMarius** [comment]
@sebastian-peter I think this discussion can be closed, since we check the uniqueness of participants. This is done with the `UniquenessValidationUtils` , which is called in the `GridContainerValidationUtils`.

**Reply by sebastian-peter** [reply]
And, for the sake of completeness: We check for uniqueness of the id field there (at least for AssetInputs).
